### PR TITLE
Add recipe for TiCC Utils

### PR DESCRIPTION
**Pull request by egpbos**
Ready for review @sodre @conda-forge/help-c-cpp

### CODE REVIEW

**Comment by egpbos** [comment]
Sorry, this package depends on the libtar package which should be added first, https://github.com/conda-forge/staged-recipes/pull/7232

**Comment by egpbos** [comment]
Libtar (#7232) is merged. When it is built (https://github.com/conda-forge/libtar-feedstock), the CI checks here should be rerun. Is there any way to trigger this @ocefpaf @sodre?

**Comment by ocefpaf** [comment]
> Libtar (#7232) is merged. When it is built ([conda-forge/libtar-feedstock](https://github.com/conda-forge/libtar-feedstock)), the CI checks here should be rerun. Is there any way to trigger this @ocefpaf @sodre?

Sometimes closing and opening it works, sometimes you'll have to send an empty commit.

**Comment by egpbos** [comment]
Seems like close/open only triggered appveyor and travis, not circleci :)

If anything pops up in a review we can avoid an empty commit.

@conda-forge/staged-recipes ready for review.

File: recipes/ticcutils/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  license_family: GPL
+  license_file: COPYING
+  summary: 'A generic utility library shared by Tilburg centre for Cognition and Communication (TiCC) software'
+  dev_url: https://github.com/LanguageMachines/ticcutils
+
+extra:
+  recipe-maintainers:
+    - egpbos
+    - jvdzwaan
+    - martinreynaert
```
Comment (scopatz) [inline recipes/ticcutils/meta.yaml]: @jvdzwaan @martinreynaert are you OK with being maintainers

Comment (martinreynaert) [reply recipes/ticcutils/meta.yaml]: I am OK with helping to OK releases. Do not expect me to actively help and develop this, please.

At this point in time I am not sure I really get what is being built here, or exactly how it is to further TICCLAT development, to be honest.

**Comment by scopatz** [comment]
There is something weird going on here with the package resolution.  Are there conflicting packages trying to be installed?

File: recipes/ticcutils/meta.yaml (hunk 1 above)
Comment (jvdzwaan) [reply recipes/ticcutils/meta.yaml]: Yes, OK with being a maintainer.

**Comment by egpbos** [comment]
It seems like there's two issues: on macOS libboost_regex cannot be found (even though it is installed in the host env) and on Linux there's something odd going on with autoconf-archive. I will try to fix the first problem first, because my local machine is a Mac as well. I already found out why I didn't notice the problem locally: `configure` was finding my Macports installed boost regex, not the Conda installed one. I will see how I can get it to find the Conda installed one instead.

**Comment by egpbos** [comment]
Weird indeed. I apparently still had built packages on my machine. A `conda build purge-all` fixed that. Now I'm also getting "Unsatisfiable dependencies for platform osx-64" for `autoconf-archive`.

**Comment by egpbos** [comment]
IIUC, the following should produce the same combination of dependencies as the build process (though in just one regular Conda environment instead of two conda-build ones (build + host)):

```sh
conda install clangxx_osx-64 autoconf autoconf-archive automake libtool pkg-config libtar libxml2 icu boost zlib bzip2 -c conda-forge
```

This works fine. I don't understand why this gives conflicts in the conda-build situation.

Could the issue be that `autoconf-archive` has `autoconf` both as a run and a build requirement? Or perhaps something with `libtar` is causing issues, although I can't see how, since it only has build requirements, no host or run ones. Perhaps it has implicit dependencies that clash with those from `ticcutils`?

For reference, these are the package requirements:
```yaml
  build:
    - {{ compiler('cxx') }}
    - autoconf
    - autoconf-archive
    - automake
    - libtool
    - pkg-config
  host:
    - libtar
    - libxml2
    - icu
    - boost
    - zlib
    - bzip2
  run:
    - libtar
    - libxml2
    - icu
    - boost
    - zlib
    - bzip2
```

**Comment by egpbos** [comment]
Aha, the `autoconf-archive` problem can be solved by adding conda-forge to my `.condarc` channels. I would expect conda-forge also to be present in the `.condarc` channels list on the CI, is this not the case?

**Comment by egpbos** [comment]
Tried switching around the channel order (defaults vs conda-forge), but in both cases, I get the same result: Boost::Regex library not found (I saw some discussion on differences between the defaults and conda-forge Boost packages, but apparently these don't matter for this case).

**Comment by egpbos** [comment]
Oh, I only now realize that in fact the Circle CI (Linux) build was not retriggered! Indeed, libtar was not yet built at that time, so it makes sense it couldn't find it. The autoconf-archive is still a bit mysterious, but I'm working on a new commit (after testing it locally), so that will retrigger it, let's see if that problem magically disappears then.

**Comment by egpbos** [comment]
Ok, so now it builds, but the make check fails on Linux... I will investigate further at the library side itself.

**Comment by egpbos** [comment]
Switching to GCC 7 fixed the tests. The problem was due to the use of regex, which was not properly supported in GCC 4.8.

As far as I'm concerned: ready to merge! @conda-forge/staged-recipes

**Comment by egpbos** [comment]
@conda-forge/staged-recipes ready for final review / merging.

**Pull request merged**